tfhe/src/integer/ciphertext/mod.rs

```diff
@@ -74,6 +75,18 @@ impl RadixCiphertext {
     pub fn block_carries_are_empty(&self) -> bool {
         self.blocks.iter().all(|block| block.carry_is_empty())
     }
+
+    /// Returns wether the ciphertext _seems_ like it holds/encrypts
+    /// a boolean (0 or 1) value.
+    ///
+    /// Since it uses degree to do so, it will not
+    /// always return the correct answer.
+    pub(crate) fn holds_boolean_value(&self) -> bool {
+        self.blocks[0].degree <= Degree(1)
+            && self.blocks[1..]
+                .iter()
+                .all(|block| block.degree == Degree(0))
+    }
```

Comment:
self.blocks[0].degree.0 ?

block.degree.0 ?

I think the first check should be == ? otherwise it's a trivial 0

otherwise return an enum ?

TrivialZero
Boolean
Integer

Reply:
A trivial zero is still a boolean value :smirk: :smirk_cat:

Reply:
I agree 😬